### PR TITLE
[16.0][ADD] stock_picking_batch_report

### PR DESCRIPTION
**Pull request by ChrisOForgeFlow**
@ForgeFlow

Original added by @Chanakya-SerpentCS, this is a port and proposition to discuss and add to OCA modules

cc. @JordiBForgeFlow

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@ChrisOForgeFlow please rebase, copier was updated

**Comment by ChrisOForgeFlow** [comment]
> @ChrisOForgeFlow please rebase, copier was updated

Done!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
Please squash last two commits.

**Comment by ChrisOForgeFlow** [comment]
> Please squash last two commits.

Done!

File: stock_picking_batch_report/models/stock_picking_batch.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            # if (
+            # rec.picking_type_id.warehouse_id
+            # and rec.picking_type_id.warehouse_id.delivery_steps != "pick_ship"
+            #             ):
+            #                 raise UserError(
+            #                         _(
+            #                             "This report is only available for "
+            #                    "warehouses configured with 2-steps delivery"
+            #                         )
+            #                     )
```
Comment (JordiMForgeFlow) [inline stock_picking_batch_report/models/stock_picking_batch.py]: maybe we could remove this?

Comment (ChrisOForgeFlow) [reply stock_picking_batch_report/models/stock_picking_batch.py]: Done!

**Review by JordiMForgeFlow — APPROVED** [review]

**Review by hparfr — APPROVED** [review]

**Pull request merged**